src/nix/flake.nix

```diff
@@ -50,6 +50,26 @@
 
             nativeBuildInputs = nixpkgsInput;
             buildInputs = nixpkgsInput;
+
+            preConfigurePhases = [
+              "setTokioMinCoreCount"
+            ];
```

Comment:
Since we're using coldsnap as a library in dropkick itself, I don't think setting the environment in the Nix flake will do anything (unless you've tested this works? in which case I have several more questions).

What I think needs to be done is replacing the `#[tokio::main]` attribute macro in `src/main.rs` with one that sets up the runtime manually, configuring the worker threads to be at a minimum of 4. (https://docs.rs/tokio/1.37.0/tokio/attr.main.html#set-number-of-worker-threads details what the equivalent code is; keep in mind we can't use the attr macro here because we probably don't want it to be static.) Then `dropkick upload` will be running coldsnap's library functions under that runtime.

Reply:
> Since we're using coldsnap as a library in dropkick itself, I don't think setting the environment in the Nix flake will do anything (unless you've tested this works? in which case I have several more questions). This will set `TOKIO_WORKER_THREADS` while building the underlying service binary (in your case, permslip).
> 
> What I think needs to be done is replacing the `#[tokio::main]` attribute macro in `src/main.rs` with one that sets up the runtime manually, configuring the worker threads to be at a minimum of 4. (https://docs.rs/tokio/1.37.0/tokio/attr.main.html#set-number-of-worker-threads details what the equivalent code is; keep in mind we can't use the attr macro here because we probably don't want it to be static.) Then `dropkick upload` will be running coldsnap's library functions under that runtime.

Right, yeah, that makes sense.. I too have questions, and am wondering if the recent successes that I had were more transient and not related then? 🤔 I think I'll try another deploy without this change.